src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -2443,105 +2442,41 @@ internal void EnsureAnonymousTypeTemplates(CancellationToken cancellationToken)
-                    {
-                        Debug.Assert(metadataDiagnostics.HasAnyErrors());
-                        return null;
-                    }
-
-                    // If the current strong name provider is the Desktop version, signing can only be done to on-disk files. 
-                    // If this binary is configured to be signed, create a temp file, output to that
-                    // then stream that to the stream that this method was called with. Otherwise output to the
-                    // stream that this method was called with.
-                    Stream retStream;
-                    if (!metadataOnly && IsRealSigned && this.Options.StrongNameProvider.Capability == SigningCapability.SignsStream)
-                    {
-                        Debug.Assert(Options.StrongNameProvider != null);
-
-                        try
-                        {
-                            signingInputStream = Options.StrongNameProvider.CreateInputStream();
-                        }
-                        catch (IOException e)
-                        {
-                            throw new Cci.PeWritingException(e);
-                        }
-
-                        retStream = signingInputStream;
-                    }
-                    else
-                    {
-                        signingInputStream = null;
-                        retStream = peStream;
-                    }
-
-                    return retStream;
+                    Stream ret;
+                    (peStream, signingInputStream, ret) = GetPeStream(metadataDiagnostics, peStreamProvider, metadataOnly);
+                    return ret;
                 };
 
-                Func<Stream> getRefPeStream;
-                if (metadataPEStreamProvider != null)
+                Func<Stream> getRefPeStream = () =>
                 {
-                    getRefPeStream = () =>
-                    {
-                        if (metadataDiagnostics.HasAnyErrors())
-                        {
-                            return null;
-                        }
+                    return GetAuxPeStream(metadataDiagnostics, metadataPEStreamProvider);
+                };
 
-                        refPeStream = metadataPEStreamProvider.GetOrCreateStream(metadataDiagnostics);
-                        Debug.Assert(refPeStream != null || metadataDiagnostics.HasAnyErrors());
-                        return refPeStream;
-                    };
-                }
-                else
+                var includePortablePdbStream = moduleBeingBuilt.DebugInformationFormat == DebugInformationFormat.PortablePdb && pdbStreamProvider != null;
+                Func<Stream> getPortablePdbStream = () =>
                 {
-                    getRefPeStream = null;
-                }
+                    portablePdbStream = GetAuxPeStream(metadataDiagnostics, pdbStreamProvider);
+                    return portablePdbStream;
+                };
```

Comment:
Are the streams returned from `getRefPeStream` and `getPortablePdbStream` disposed?

Reply:
It looks like they aren't, which is odd because this was true in the old code as well.  I've added a few `using`s for both.